reps/2022-08-31-actor-affinity-apis.md

```diff
@@ -0,0 +1,420 @@
+
+## Summary
+### General Motivation
+
+Provides a set of lightweight actor affinity and anti-affinity scheduling interfaces.
+Replacing the heavy PG interface to implement collocate or spread actors.
+
+* Affinity
+  * Co-locate the actors in the same node.
+  * Co-locate the actors in the same batch of nodes, like nodes in the same zones
+* Anti-affinity
+  * Spread the actors of a service across nodes and/or availability zones, e.g. to reduce correlated failures.
+  * Give a actor "exclusive" access to a node to guarantee resource isolation 
+  * Spread the actors of different services that will affect each other on different nodes.
+
+
+### Should this change be within `ray` or outside?
+
+Yes, this will be a complement to ray core's ability to flexibly schedule actors/tasks.
+
+## Stewardship
+### Required Reviewers
+@WangTaoTheTonic
+### Shepherd of the Proposal (should be a senior committer)
+
+
+## Design and Architecture
+
+### Brief idea
+
+1. Tag the Actor with key-value labels first
```

Comment:
Could we apply this API to node as well?

Reply:
Yes. Node Affinity can also be implemented this way. I have the same consideration later. But it needs to be done after ActorAffinity feature with raylet scheduling mode  completed.